### ISSUE TITLE
Genelizability issue

### ISSUE THREAD

**Issue by Woody-Wan** [chenoly/CRMark#3]
Thanks for sharing and maintaining this repo! I have been training using train_more.py on mirflickr25k dataset, and I find that the training is not very stable and often gets Nan loss after several epochs. Is author aware of this Nan issue before? 

To provide more context, before updating the watermarklab functions, I have been using level 1 imagenet c to train the model, and mse as secret loss seems to perform worse than bce. However, even with some tuning, the accuracy hardly gets 99% after adding augmentation.

**Comment by chenoly** [comment]
NaN is a common issue in INN-based data hiding methods training, often stemming from factors such as network architecture design and the configuration of distortion layers. For different datasets, it is often necessary to fine-tune the relevant parameters within the dynamic loss weighting strategy to ensure stable training. I hope these insights are helpful to you!
________________________________
发件人: Woody-Wan ***@***.***>
发送时间: 2026年3月5日 14:15
收件人: chenoly/CRMark ***@***.***>
抄送: Subscribed ***@***.***>
主题: [chenoly/CRMark] Genelizability issue (Issue #3)

[https://avatars.githubusercontent.com/u/63642698?s=20&v=4]Woody-Wan created an issue (chenoly/CRMark#3)<https://github.com/chenoly/CRMark/issues/3>

Thanks for sharing and maintain this repo! I have been training using train_more.py on mirflickr25k dataset, and I find that the training is not very stable and often get Nan loss after several epochs. Is author aware of this Nan issue before?

To provide more context, before updating the watermarklab functions, I have been using level 1 imagenet c to train the model, and mse as secret loss seems to perform worse than bce. However, even with some tuning, the accuracy hardly gets 99% after adding augmentation.

―
